pilot/pkg/xds/ep_filters.go

```diff
@@ -138,6 +155,24 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 	return filtered
 }
 
+func (b *EndpointBuilder) selectNetworkGateways(nw network.ID, c cluster.ID) []*model.NetworkGateway {
```

Comment:
This is a big improvement – can you add a comment explaining the benefits:
* Potentially reduce hops
* MCS case